lib/portlayer/store/store.go

```diff
@@ -60,16 +66,26 @@ var (
 func Init(ctx context.Context, session *session.Session, imgStoreURL url.URL) error {
 	defer trace.End(trace.Begin(imgStoreURL.String()))
 
-	mgr = &StoreManager{
-		dsStores:     make(map[string]*kvstore.KeyValueStore),
-		datastoreURL: imgStoreURL,
-	}
-	//create or restore the api accessible datastore backed k/v store
-	return NewDatastoreKeyValue(ctx, session, APIKV)
+	func() error {
```

Comment:
Not following why this idiom is preferred.  I feel like I've asked this before so apologies if I have.  A comment at the least would be appreciated.